src/GafferOSL/OSLShader.cpp

```diff
@@ -84,9 +104,20 @@ struct ShadingEngineCacheGetterKey
 	{
 	}
 
-	ShadingEngineCacheGetterKey( const OSLShader *s )
-		:	shader( s ), hash( s->attributesHash() )
+	ShadingEngineCacheGetterKey( const OSLShader *s, const CompoundObject *targetA )
+		:	shader( s ), targetAttributes( targetA )
 	{
+		hash = s->attributesHash();
+		if( targetAttributes )
+		{
+			
+			ConstCompoundObjectPtr attributes = shader->attributes();
+			const ShaderNetwork *network = findOSLShaderNetwork( attributes.get() );
+			if( network && network->size() )
+			{
+				network->hashSubstitutions( targetAttributes, hash );
```

Comment:
Since we're going out of our way to "deduplicate" ShadingEngines as much as possible here, it'd be good to have some test coverage for that.

Reply:
Is this the sort of thing you were thinking of? e3df5be

Reply:
Yep, exactly!